### PR TITLE
Edit documentation to show all examples within `if (require("maps"))` statements

### LINKED ISSUE #3064: add `plot(last_plot())` in function examples to make pkgdown website more vivid
pkgdown's biggest feature is including output of examples, which is especially useful for ggplot.

But only plots which are automatically printed can be displayed in the pkgdown website. For some cases, like https://ggplot2.tidyverse.org/reference/borders.html

```r
if (require("maps")) {

    ggplot() + ...

    ggplot() + ... + ...

    ggplot() + ... + ...
}
```

only the last plot (returned by `if` statement) is displayed
image

### PR DESCRIPTION
**Pull request by sowla**
Fixes #3064

- show all examples in fortify-map.r, annotation-map.r, coord-map.r (using a lot of if statements instead of the suggested `plot(last_plot())` so plots appear with code)
- fix bug in Iowa example

### CODE REVIEW

File: R/fortify-map.r [hunk 1]
```diff
@@ -97,20 +101,25 @@ map_data <- function(map, region = ".", exact = FALSE, ...) {
 #' ia <- map_data("county", "iowa")
 #' mid_range <- function(x) mean(range(x))
 #' seats <- do.call(rbind, lapply(split(ia, ia$subregion), function(d) {
-#'   data.frame(lat = mid_range(d$lat), long = mid_range(d$long))
+#'     data.frame(lat = mid_range(d$lat), long = mid_range(d$long), subregion = unique(d$subregion))
```
Comment (thomasp85) [inline R/fortify-map.r]: Any reason for this addition? also, make sure to keep the 2-space indentation

Comment (karawoo) [reply R/fortify-map.r]: Yes this was a bug in the example code, `subregion` is used for the label aesthetic but wasn't a column in the data.

Comment (thomasp85) [reply R/fortify-map.r]: Ah, great

Comment (thomasp85) [reply R/fortify-map.r]: but the indentation should be fixed

Comment (sowla) [reply R/fortify-map.r]: oops.. thanks!

File: R/fortify-map.r [hunk 2]
```diff
@@ -65,11 +67,13 @@ fortify.map <- function(model, data, ...) {
...
 #'   geom_polygon(aes(group = group, fill = assault)) +
 #'   coord_map("albers",  at0 = 45.5, lat1 = 29.5)
+#' }
 #'
+#' if (require("maps")) {
 #' ggplot(choro, aes(long, lat)) +
 #'   geom_polygon(aes(group = group, fill = assault / murder)) +
 #'   coord_map("albers",  at0 = 45.5, lat1 = 29.5)
-#' }
+#'  }
```
Comment (thomasp85) [inline R/fortify-map.r]: some whitespace changes crept in here

File: R/fortify-map.r [hunk 3]
```diff
@@ -97,20 +101,25 @@ map_data <- function(map, region = ".", exact = FALSE, ...) {
 #' ia <- map_data("county", "iowa")
 #' mid_range <- function(x) mean(range(x))
 #' seats <- do.call(rbind, lapply(split(ia, ia$subregion), function(d) {
-#'   data.frame(lat = mid_range(d$lat), long = mid_range(d$long))
+#'  data.frame(lat = mid_range(d$lat), long = mid_range(d$long), subregion = unique(d$subregion))
```
Comment (thomasp85) [inline R/fortify-map.r]: needs an additional space

**Comment by sowla** [comment]
I'm so sorry, I forgot to diff before committing.. this should be it now I hope! :)

**Pull request merged**